### PR TITLE
566 password recovery

### PR DESCRIPTION
**Pull request by RamyAlshurafa**
#### Issues this PR relates to
#566 
---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install
add this to the backend .env file
```
RESET_TOKEN_MAX_AGE = 3600000
```

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars
- `npm i`
- `npm run build:db`

**to test the email   comment out [this if statement](https://github.com/yalla-coop/tempo/blob/d3d9bf8955668e896b83413e1549a8f279a3c0dc/server/services/mailing/index.js#L45-L46)**

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/pages/ForgetPassword/index.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    resetPasswordLink,
+    sendLinkSuccess,
+    sendLinkLoading,
+    sendLinkError,
+  } = props;
+
+  const handleEmail = (_email) => {
+    if (attemptToSubmit) {
+      const isValidEmail = emailSchema.isValidSync(_email);
+      setEmailError(isValidEmail ? "" : "invalid email");
```
Comment (thejoefriel) [inline client/src/pages/ForgetPassword/index.js:30]: Change error message to "Invalid email"

File: client/src/pages/ForgetPassword/index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    setEmail(_email);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    setAttemptToSubmit(true);
+    const isValidEmail = emailSchema.isValidSync(email);
+    setEmailError(isValidEmail ? "" : "invalid email");
```
Comment (thejoefriel) [inline client/src/pages/ForgetPassword/index.js:40]: Same here re error message. Just make sure all error messages start with a capital letter

File: client/src/pages/ForgetPassword/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              type="email"
+              placeholder="Type email here..."
+              label="Email"
+              mt="5"
+              error={emailError || sendLinkError.message}
+            />
+            <Button
+              loading={sendLinkLoading}
+              disabled={!email || !!emailError || sendLinkLoading}
+              style={{ marginTop: "71px" }}
```
Comment (thejoefriel) [inline client/src/pages/ForgetPassword/index.js:72]: I believe on mobile this should be 40px and on desktop make it 70px. There seems to be different margin tops on the other screens as well. This is probably a mistake in wireframes not being checked closely enough, so please make it consistent of 70px desktop (40px mobile).

File: server/services/errorHandler/messages.js [hunk 4]
```diff
@@ -92,6 +92,22 @@ const errors = {
...
+    new CError({
+      message: 'email not found',
+      httpStatusCode: 404,
+      data,
+    }),
+  RESET_PASSWORD_TOKEN_INVALID_OR_EXPIRED: (
+    data, // on set password
+  ) =>
+    new CError({
+      message: 'token invalid or expired',
```
Comment (thejoefriel) [inline server/services/errorHandler/messages.js:107]: Change to "Your link has expired or is invalid"

Comment (RamyAlshurafa) [reply server/services/errorHandler/messages.js:107]: done

File: client/src/pages/ForgetPassword/index.js (hunk 2 above)
Comment (RamyAlshurafa) [reply client/src/pages/ForgetPassword/index.js:40]: done

File: client/src/pages/ForgetPassword/index.js (hunk 1 above)
Comment (RamyAlshurafa) [reply client/src/pages/ForgetPassword/index.js:30]: done

File: client/src/pages/ForgetPassword/index.js (hunk 3 above)
Comment (RamyAlshurafa) [reply client/src/pages/ForgetPassword/index.js:72]: done

**Comment by RamyAlshurafa** [comment]
Thanks @thejoefriel 
I wasn't able to fix the extra space above the graphic as it comes from the layout container, and i dont want to change it globally, so 
- if this page dont follow the global layout i can create new layout for it
- if the layout of all pages is changing on this release we can leave this and open an issue to fix the pages layouts 
what do you think?

**Comment by thejoefriel** [comment]
> Thanks @thejoefriel
> I wasn't able to fix the extra space above the graphic as it comes from the layout container, and i dont want to change it globally, so
> 
> * if this page dont follow the global layout i can create new layout for it
> * if the layout of all pages is changing on this release we can leave this and open an issue to fix the pages layouts
>   what do you think?

Yeah I agree, can you create a separate issue for now please and I can look at it later :)

**Pull request merged**